sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -131,6 +133,13 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
         } else if (options.inferTimestamp &&
             (allCatch opt timestampFormatter.parse(field)).isDefined) {
           TimestampType
+        } else if (options.allowDateFormat &&
+          (allCatch opt { val dtFormat = new SimpleDateFormat(
```

Comment:
Can we avoid using `SimpleDateFormat`, and trying date type (smaller type) first before timestamp type?

Reply:
done